kube-core/src/schema/transform_any_of.rs

```diff
@@ -0,0 +1,238 @@
+use std::ops::DerefMut;
+
+use schemars::transform::Transform;
+
+use crate::schema::{Schema, SchemaObject, SubschemaValidation, NULL_SCHEMA};
+
+/// Replace the schema with the anyOf subschema and set to nullable when the
+/// only other subschema is the nullable entry.
+///
+/// Used for correcting the schema for optional tagged unit enums.
+/// The non-null subschema is hoisted, and nullable will be set to true.
+///
+/// This will return early without modifications unless:
+/// - There are exactly 2 `anyOf` subschemas.
+/// - One subschema represents the nullability (ie: it has an `enum` with a single
+///   `null` entry, and `nullable` set to true).
+///
+/// NOTE: This should work regardless of whether other hoisting has been performed or not.
+#[derive(Debug, Clone)]
+pub struct AnyOfSchemaRewriter;
```

Comment:
I believe that this method has no functional differences with implementation in https://github.com/kube-rs/kube/pull/1853 and therefore will pass the tests. We just try to panic more, where we should not.